sparse/file.go

```diff
@@ -42,18 +45,32 @@ func NewBufferedFileIoProcessor(name string, flag int, perm os.FileMode, isCreat
 		return nil, err
 	}
 
-	return &BufferedFileIoProcessor{file}, nil
+	return &BufferedFileIoProcessor{
+		File:   file,
+		fieMap: NewFiemapFile(file),
+	}, nil
 }
 
 func NewBufferedFileIoProcessorByFP(fp *os.File) *BufferedFileIoProcessor {
-	return &BufferedFileIoProcessor{fp}
+	return &BufferedFileIoProcessor{
+		File:   fp,
+		fieMap: NewFiemapFile(fp),
+	}
+}
+
+func (file *BufferedFileIoProcessor) UnmapAt(length uint32, offset int64) (int, error) {
+	return int(length), file.fieMap.PunchHole(offset, int64(length))
 }
 
 func (file *BufferedFileIoProcessor) GetFile() *os.File {
 	return file.File
 }
 
-func (file *BufferedFileIoProcessor) GetDataLayout (ctx context.Context) (<-chan FileInterval, <-chan error, error) {
+func (file *BufferedFileIoProcessor) GetFieMap() FiemapFile {
```

Comment:
Suggest returning the reference `*FiemapFile` instead of creating another structure.

Also, it seems we can refactor `file.File` to private `file.file` in BufferedFileIoProcessor and DirectFileIoProcessor, because it seems all of them just be exported by `GetFile` method.